### DISCUSSION TITLE
Accessing autoranged chart axis bounds

### DISCUSSION THREAD

**Discussion by juhoimmonen** [scalafx/scalafx#338] [Q&A]
Hi!

I'm plotting my original data scattered and regression model in a line to a line chart. 

By creating the empty chart x-axis is set from 0 to 110. After plotting my data x ranging from 1 to 5 the axis is adjusted from 0 to 6. How can I access these autoranged lower and upper bounds? xAxis.getUpperBound gives the original default of 110.

**Comment by jpsacha** [comment]
Can you provide some sample code to illustrate your question?

**Reply by juhoimmonen** [reply]
Sure, thanks! 

// Formulate an empty chart:

// Axes:
val xAxis = new NumberAxis()
val yAxis = new NumberAxis()

// Formulate empty series for the scatter plot:
val scatterData: ObservableBuffer[(Double, Double)] = ObservableBuffer()
val scatterXY = scatterData.map(x => XYChart.Data[Number, Number](x._1, x._2))
val scatterSeries = XYChart.Series("Data", scatterXY)

// Draw the chart:
val lineChart = LineChart(xAxis, yAxis, ObservableBuffer(scatterSeries))

// Update scatter data with data x ranging from 1 to 5, for example
scatterXY.appendAll(data)

// xAxis upper bound is the original 110
println(xAxis.getUpperBound)

**Reply by jpsacha** [reply]
You are likely querying the value of the `upperBound` before it had chance to change. Add an observer and you should see when it is updated:

```scala
  xAxis.upperBound.onChange { (source, oldValue, newValue) =>
    println(s"source:$source,  oldValue:$oldValue,  newValue:$newValue")
  }
```

**Reply by juhoimmonen** [reply]
Thank you, obviously it was about awaiting. Could have naturally run my code also better first.
